README.md

```diff
@@ -1,15 +1,121 @@
+This repo contains contracts, deployment, and SDK tooling for Hyperlane Warp Routes. 
 
-## Versions
+## Warp Route Architecture
 
-| Git Ref | Release Date | Notes |
-| ------- | ------------ | ----- |
-| [audit-v2-remediation]() | 2023-02-15 | Hyperlane V2 Audit remediation |
-| [main]() | ~ | Bleeding edge |
+A *warp route* is a collection of [`TokenRouter`](./contracts/libs/TokenRouter.sol) contracts deployed across a set of Hyperlane chains. These contracts leverage the `Router` pattern to implement access control and routing logic for remote token transfers. These contracts send and receive [`Message`](./contracts/libs/Message.sol)s which encode payloads containing a transfer `amount` and `recipient` address.
+
+```mermaid
+%%{ init: {
+  "theme": "neutral",
+  "themeVariables": {
+    "mainBkg": "#025AA1",
+    "textColor": "white",
+    "clusterBkg": "beige"
+  },
+  "themeCSS": ".edgeLabel { color: black }"
+}}%%
+
+graph LR
+    subgraph "Ethereum"
+        HYP_E[TokenRouter]
+        style HYP_E fill:purple
+        Mailbox_E[(Mailbox)]
+    end
+
+    subgraph "Polygon"
+        HYP_P[TokenRouter]
+        style HYP_P fill:purple
+        Mailbox_P[(Mailbox)]
+    end
+
+
+    subgraph "Gnosis"
+        HYP_G[TokenRouter]
+        style HYP_G fill:purple
+        Mailbox_G[(Mailbox)]
+    end
+
+    HYP_E -. "router" .- HYP_P -. "router" .- HYP_G
+
+```
+
+The Token Router contract comes in several flavors and a warp route can be composed as  a combination of these flavors.
+
+- [`Native`](./contracts/HypNative.sol) - for warping native assets (e.g. ETH) from the canonical chain
+- [`Collateral`](./contracts/HypERC20Collateral.sol) - for warping tokens, ERC20 or ERC721, from the canonical chain
+- [`Synthetic`](./contracts/HypERC20.sol) - for representing tokens, Native/ERC20 or ERC721, on a non-canonical chain
+
+## Remote Transfer Lifecycle Diagrams
+
+To initiate a remote transfer, users call the `TokenRouter.transferRemote` function with the destination chain ID, recipient address, and transfer amount. 
+
+<details>
+<summary>Native Remote Transfer</summary>
+
+### Transfer Alice's `amount` native ETH from Ethereum to Bob on Polygon
+
+```mermaid
+%%{ init: {
+  "theme": "neutral",
+  "themeVariables": {
+    "mainBkg": "#025AA1",
+    "textColor": "white",
+    "clusterBkg": "beige"
+  },
+  "themeCSS": ".edgeLabel { color: black }"
+}}%%
+
+graph TB
+    Bob((Bob))
+    style Bob fill:black
+    Alice((Alice))
+    style Alice fill:black
+
+    Relayer([Relayer])
+
+    subgraph "Ethereum"
+        HYP_E[NativeTokenRouter]
+        IGP_E[InterchainGasPaymaster]
```

Comment:
I would personally omit IGP, Relayers here. Maybe even ISMs and just point folks to SC/ISM documentation